core_data_modules/traced_data/traced_data.py

```diff
@@ -317,6 +321,24 @@ def get_history(self, key):
         history.sort(key=lambda x: x["timestamp"])
         return history
 
+    def clear_history(self, user, file_sha):
```

Comment:
I think we make it clear that this is an internal method that shouldn't ever be called directly within the pipeline.  I think the signal for this is to name it `__clear_history`?

Reply:
I definitely don't think this should be public - I wouldn't want pipelines calling this directly. However, private also wasn't perfect because it needs to be called from TracedDataJsonIO. Ideally I would be able to mark it as "package private", but on balance I think I should go with private and accept the warning in TracedDataJsonIO.

Reply:
Made "private" with a justifying comment at its usage in TracedDataJsonIO